sound/soc/sof/intel/hda-trace.c

```diff
@@ -44,6 +44,11 @@ int hda_dsp_trace_init(struct snd_sof_dev *sdev, u32 *tag)
 		(struct sof_intel_hda_dev *)sdev->pdata->hw_pdata;
 	int ret;
 
+	if (!hda) {
+		dev_err(sdev->dev, "error: forget to assign hw_pdata?\n");
+		return -ENODEV;
+	}
+
```

Comment:
I don't get why this is needed? the trace_init can only happen if the probe is successful, no?

Reply:
@bardliao can we not argue the same for the first commit?

Reply:
@ranj063 Sorry, I didn't get your question. I didn't argue with anything. I was trying to explain the second patch. But I don't insist on it. It's fine for me if the second patch is not applied. :)

Reply:
@bardliao sorry please dont take it personally. All i was trying to say was that the situation that you described with .probe being somthing other than hda_dsp_probe() could lead to the same problem in hda_dsp_remove() just like in hda_dsp_trace_init().

Reply:
@ranj063 Oh yes, indeed, there is also the same risk in the first patch.  However, in my opinion, the .probe and .remove ops are a pair. That's why I didn't think set .probe = hda_dsp_probe() but not .remove = hda_dsp_remove()  will happen in real world. But I agree with you, we should use the same criterion for both patches. Lets revert the second patch.